include/bluetooth/gatt.h

```diff
@@ -1340,13 +1356,27 @@ struct bt_gatt_discover_params {
  *  For each attribute found the callback is called which can then decide
  *  whether to continue discovering or stop.
  *
- *  @note This procedure is asynchronous therefore the parameters need to
- *        remains valid while it is active.
+ *  The Response comes in callback @p params->func. The callback is run from
```

Comment:
We could potentially use something like pinctrl here, its own "named" section: https://docs.zephyrproject.org/latest/doxygen/html/group__pinctrl__interface.html#ga46d7997af7e9231b431c79af34288d50